### PR TITLE
Ticket1518 file watcher error

### PR DESCRIPTION
**Pull request by AdrianPotter**
### Description of work

_Wait for https://github.com/ISISComputingGroup/EPICS-inst_servers/pull/80 to be merged_

The filewatcher is supposed to be disabled when the block server writes to the file system. However in the previous setup, the asynchronous nature of the write queue meant that often (e.g. synoptics), the filewatcher was re-enabled before the command was executed.

I've changed the flow of logic so that the filewatcher is disabled as part of the execution of the write queue so it will automatically apply to all write commands.

Note that if a write command is executed outside of the queue, as is currently the case for config methods, then they will now throw the error currently reported by the synoptics. This will be fixed by ticket #1500. Please wait for that to be merged first.

Also tidied up some of the exception coding to Python standards.
### To test

https://github.com/ISISComputingGroup/IBEX/issues/1518
### Acceptance criteria
- [ ] Deleting synoptics no longer results in `filewtchr` errors
- [ ] Add/edit/remove operations for configs/components/synoptics continue to work as expected with no `filewtchr` errors
- [ ] Modifying/deleting files from the file system directly still result in errors from the `filewtchr` and the file is restored automatically by the block server.

---
#### Code Review
- [ ] Is the code of an acceptable quality?
- [ ] Has the author taken into account the multi-thredded nature of the code?
- [ ] Have the changes been documented in the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev). If so, do they describe the changes appropriately?
- [ ] Has the [manual system tests spreadsheet](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Manual-system-tests) been updated?
### Functional Tests
- [ ] Do changes function as described? Add comments below that describe the tests performed.
### Final steps
- [ ] Reviewer has updated the submodule in the main EPICS repo? See **Reviewing work for the subModules of EPICS** in the [Git workflow](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Git-workflow) page for details.
- [ ] Reviewer has moved the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev) entry for this ticket in the "Changes merged into master" section

### CODE REVIEW

File: block_server.py [hunk 1]
```diff
@@ -319,6 +319,9 @@ def write(self, reason, value):
         """A method called by SimpleServer when a PV is written to the BlockServer over Channel Access. The write
             commands are queued as Channel Access is single-threaded.
 
+            Note that the filewatcher is disabled as part of the write queue so any operations that intend to modify
+            files should use the write queue. See tickets 1500 and 1518
```
Comment (John-Holt-Tessella) [inline block_server.py]: I don't think referring to the tickets helps here.

File: block_server.py [hunk 2]
```diff
@@ -664,15 +657,17 @@ def consume_write_queue(self):
             self.load_config, ("configname",), "LOADING_CONFIG")
         """
         while True:
-            while len(self.write_queue) > 0:
-                with self.write_lock:
+            with self.write_lock:
+                if self._filewatcher is not None: self._filewatcher.pause()
```
Comment (John-Holt-Tessella) [inline block_server.py]: Pop the item off the queue with the write lock then release the lock and process the command.

File: block_server.py [hunk 3]
```diff
@@ -664,15 +657,17 @@ def consume_write_queue(self):
...
+                while len(self.write_queue) > 0:
                     cmd, arg, state = self.write_queue.pop(0)
```
Comment (John-Holt-Tessella) [inline block_server.py]: What happens if this throws and exception. Could you check please.

File: block_server.py (hunk 2 above)
Comment (AdrianPotter) [inline block_server.py]: I think the original reason I did it this way was that I was concerned about constantly pausing and resuming the filewatcher, but the suggested way is definitely better. Changed 	52c0827

File: block_server.py (hunk 3 above)
Comment (AdrianPotter) [inline block_server.py]: Exception kills the block server:

```
[Fri Oct 28 10:30:48 2016] Exception in thread Thread-2:
[Fri Oct 28 10:30:48 2016] Traceback (most recent call last):
[Fri Oct 28 10:30:48 2016]   File "C:\Instrument\Apps\Python\lib\threading.py", line 810, in __bootstrap_inner
[Fri Oct 28 10:30:48 2016]     self.run()
[Fri Oct 28 10:30:48 2016]   File "C:\Instrument\Apps\Python\lib\threading.py", line 763, in run
[Fri Oct 28 10:30:48 2016]     self.__target(*self.__args, **self.__kwargs)
[Fri Oct 28 10:30:48 2016]   File "C:\Instrument\Apps\EPICS\ISIS\inst_servers\master\\block_server.py", line 672, in consume_write_queue
[Fri Oct 28 10:30:48 2016]     cmd()
[Fri Oct 28 10:30:48 2016]   File "C:\Instrument\Apps\EPICS\ISIS\inst_servers\master\\block_server.py", line 59, in neverWork
[Fri Oct 28 10:30:48 2016]     raise Exception("What's black and white and red all over?")
[Fri Oct 28 10:30:48 2016] Exception: What's black and white and red all over?
[Fri Oct 28 10:30:49 2016] 
```

Comment (AdrianPotter) [inline block_server.py]: I've added a catch for exceptions thrown by the write queue: caee13c

**Pull request merged**